apps/dashboard/app/javascript/packs/navigator/data_table.js

```diff
@@ -0,0 +1,306 @@
+                { 
+                    name: 'name', 
+                    data: 'name', 
+                    className: 'text-break',
+                    render: (data, type, row, meta) => {
+                        return `<span id='${row.id}' data="${row.url}" style="cursor: pointer;">${row.name}</a>`;
+                    }
+ 
+                },
+                { visible: false, name: 'actions', orderable: false, data: null, render: (data, type, row, meta) => this.actionsBtnTemplate({ row_index: meta.row, file: row.type != 'd', data: row }) },
+                {
+                    visible: false,
+                    data: 'size',
+                    render: (data, type, row, meta) => {
+                        return type == "display" ? row.human_size : data;
+                    }
+                }, // human_size
+                {
+                    visible: false,
+                    data: 'modified_at', render: (data, type, row, meta) => {
+                        if (type == "display") {
+                            let date = new Date(data * 1000)
+
+                            // Return formatted date "3/23/2021 10:52:28 AM"
+                            return isNaN(data) ? 'Invalid Date' : `${date.toLocaleDateString()} ${date.toLocaleTimeString()}`
+                        }
+                        else {
+                            return data;
+                        }
+                    }
+                }, // modified_at
+                { name: 'owner', data: 'owner', visible: false }, // owner
+                {
+                    name: 'mode', 
+                    data: 'mode', 
+                    visible: false,
+                } // mode
+            ]
+        });
+        
+        /*
+            These 2 checkboxes will go away once the ability to pass in options from the form is written.
+            They are here now to show functionality of hiding/show files/directories
+        */
+        $('#directory-contents_filter').prepend(
+            `<label for="show-dotfiles">
+                <input type="checkbox" id="show-dotfiles" ${this.getShowDotFiles() ? 'checked' : ''}> Show Dotfiles</label>`);
+
+        $('#directory-contents_filter').prepend(
+            `<label style="margin-right: 20px" for="show-files">
+                <input type="checkbox" id="show-files" ${this.getShowFiles() ? 'checked' : ''}> Show Files</label>`);
+        
+
+    }
+
+    async reloadTable(url) {
+        if(url) {
+            this._url = url;
+        }
+
+        var request_url = this._url;
+
+
+        if (history.state != null && url == history.state.currentDirectoryUrl) {
+            request_url = url || history.state.currentDirectoryUrl;
+        }
+
+        if(request_url) {
+            try {
+                const response = await fetch(request_url, { headers: { 'Accept': 'application/json' } });
```

Comment:
I had to patch this a few weeks ago in #2645. When it's cached it conflicts with the HTML and the browers' back button.

```suggestion
                const response = await fetch(request_url, { headers: { 'Accept': 'application/json' }, cache: 'no-store' });
```